sql/tests: TestErrorDuringExtendedProtocolCommit failed

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9543621?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9543621?buildTab=artifacts#/) on master @ [a5d61d2cf2bfc11bd6bdc27dcab32467334fb123](https://github.com/cockroachdb/cockroach/commits/a5d61d2cf2bfc11bd6bdc27dcab32467334fb123):


```
I230411 08:13:12.089340 8581 1@server/server_sql.go:1736  [T1,n1] 1020  serving sql connections
I230411 08:13:12.090322 8581 testutils/testcluster/testcluster.go:1375  [-] 1021  WaitForFullReplication
I230411 08:13:12.090353 8581 testutils/testcluster/testcluster.go:1379  [-] 1022  WaitForFullReplication took: 340ns
I230411 08:13:12.104571 10687 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-capability-watcher] 1023  Establishing MuxRangeFeed (/Table/8/{1-2}@1681200792.052620466,0...; 1 spans)
I230411 08:13:12.104753 10939 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-capability-watcher] 1024  MuxRangeFeed starting for range /Table/8/{1-2}@1681200792.052620466,0 (rangeID 12, attempt 0)
I230411 08:13:12.104842 10940 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-capability-watcher,mux_n=1,gen=2] 1025  Establishing MuxRangeFeed to node 1
I230411 08:13:12.106890 10126 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:77  [T1,n1,rangefeed=tenant-settings-watcher] 1026  Establishing MuxRangeFeed (/Table/5{0-1}@1681200792.052029517,0...; 1 spans)
I230411 08:13:12.107053 11011 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:212  [T1,n1,rangefeed=tenant-settings-watcher] 1027  MuxRangeFeed starting for range /Table/5{0-1}@1681200792.052029517,0 (rangeID 51, attempt 0)
I230411 08:13:12.107177 11012 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:297  [T1,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 1028  Establishing MuxRangeFeed to node 1
    autocommit_extended_protocol_test.go:197: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/tests/autocommit_extended_protocol_test.go:197
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
I230411 08:13:12.109918 8581 testutils/testcluster/testcluster.go:145  [-] 1029  TestCluster quiescing nodes
I230411 08:13:12.110092 9263 sql/stats/automatic_stats.go:550  [T1,n1] 1030  quiescing auto stats refresher
I230411 08:13:12.110306 4618 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=lease,mux_n=1,gen=2] 1031  MuxRangeFeed to node 1 terminating after 1.084540665s with err=recv msg error: recv: context canceled
I230411 08:13:12.110408 8970 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=lease] 1032  MuxRangeFeed  terminating after 1.084831858s with err=recv msg error: recv: context canceled
I230411 08:13:12.110375 9909 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,job=AUTO SPAN CONFIG RECONCILIATION id=855672065312325633,rangefeed=sql-watcher-protected-ts-records-rangefeed,mux_n=1,gen=2] 1033  MuxRangeFeed to node 1 terminating after 944.102402ms with err=recv msg error: recv: context canceled
I230411 08:13:12.110502 9758 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,job=AUTO SPAN CONFIG RECONCILIATION id=855672065312325633,rangefeed=sql-watcher-protected-ts-records-rangefeed] 1034  MuxRangeFeed  terminating after 944.39644ms with err=recv msg error: recv: context canceled
W230411 08:13:12.110531 10330 kv/kvserver/intentresolver/intent_resolver.go:826  [-] 1035  failed to gc transaction record: could not GC completed transaction anchored at /Table/3/1/102/2/1: node unavailable; try another peer
I230411 08:13:12.110574 8836 server/start_listen.go:103  [T1,n1] 1036  server shutting down: instructing cmux to stop accepting
I230411 08:13:12.110632 10255 rpc/context.go:2302  [T1,n1,rnode=1,raddr=127.0.0.1:33691,class=default,rpc] 1038  connection heartbeat loop ended with err: <nil>
I230411 08:13:12.110702 9674 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=sql_instances,mux_n=1,gen=2] 1039  MuxRangeFeed to node 1 terminating after 1.03597695s with err=recv msg error: recv: context canceled
I230411 08:13:12.110704 9914 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,job=AUTO SPAN CONFIG RECONCILIATION id=855672065312325633,rangefeed=sql-watcher-descriptor-rangefeed,mux_n=1,gen=2] 1041  MuxRangeFeed to node 1 terminating after 943.957818ms with err=recv msg error: recv: context canceled
W230411 08:13:12.110813 9112 sql/sqlliveness/slinstance/slinstance.go:334  [T1,n1] 1043  exiting heartbeat loop
I230411 08:13:12.110805 9671 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=sql_instances] 1044  MuxRangeFeed  terminating after 1.036225422s with err=recv msg error: recv: context canceled
W230411 08:13:12.110846 9112 sql/sqlliveness/slinstance/slinstance.go:321  [T1,n1] 1045  exiting heartbeat loop with error: node unavailable; try another peer
I230411 08:13:12.110816 9918 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,job=AUTO SPAN CONFIG RECONCILIATION id=855672065312325633,rangefeed=sql-watcher-zones-rangefeed,mux_n=1,gen=2] 1046  MuxRangeFeed to node 1 terminating after 943.798169ms with err=recv msg error: recv: context canceled
I230411 08:13:12.110743 10931 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 1042  MuxRangeFeed to node 1 terminating after 57.771869ms with err=recv msg error: recv: context canceled
I230411 08:13:12.110953 9481 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=spanconfig-subscriber,mux_n=1,gen=2] 1049  MuxRangeFeed to node 1 terminating after 1.046839269s with err=recv msg error: recv: context canceled
I230411 08:13:12.110976 10122 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=tenant-boundaries-watcher] 1051  MuxRangeFeed  terminating after 58.712697ms with err=recv msg error: recv: context canceled
I230411 08:13:12.110972 9688 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=settings-watcher,mux_n=1,gen=2] 1053  MuxRangeFeed to node 1 terminating after 1.040801268s with err=recv msg error: recv: context canceled
I230411 08:13:12.110712 9697 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=system-config-cache,mux_n=1,gen=2] 1040  MuxRangeFeed to node 1 terminating after 1.029245639s with err=recv msg error: recv: context canceled
I230411 08:13:12.111039 9478 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=spanconfig-subscriber] 1055  MuxRangeFeed  terminating after 1.047126344s with err=recv msg error: recv: context canceled
I230411 08:13:12.111071 9685 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=settings-watcher] 1056  MuxRangeFeed  terminating after 1.041063689s with err=recv msg error: recv: context canceled
I230411 08:13:12.110955 11012 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 1050  MuxRangeFeed to node 1 terminating after 3.734334ms with err=recv msg error: recv: context canceled
I230411 08:13:12.110576 10940 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=tenant-capability-watcher,mux_n=1,gen=2] 1037  MuxRangeFeed to node 1 terminating after 5.703467ms with err=recv msg error: recv: context canceled
E230411 08:13:12.110884 9112 server/server_sql.go:508  [T1,n1] 1048  failed to run update of instance with new session ID: node unavailable; try another peer
I230411 08:13:12.110955 9906 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,job=AUTO SPAN CONFIG RECONCILIATION id=855672065312325633,rangefeed=sql-watcher-zones-rangefeed] 1052  MuxRangeFeed  terminating after 944.074559ms with err=recv msg error: recv: context canceled
I230411 08:13:12.110844 9760 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,job=AUTO SPAN CONFIG RECONCILIATION id=855672065312325633,rangefeed=sql-watcher-descriptor-rangefeed] 1047  MuxRangeFeed  terminating after 944.274153ms with err=recv msg error: recv: context canceled
I230411 08:13:12.111232 10687 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=tenant-capability-watcher] 1059  MuxRangeFeed  terminating after 6.52479ms with err=recv msg error: recv: context canceled
I230411 08:13:12.110966 9057 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:300  [T1,n1,rangefeed=table-stats-cache,mux_n=1,gen=2] 1054  MuxRangeFeed to node 1 terminating after 1.084971697s with err=recv msg error: recv: context canceled
I230411 08:13:12.111295 9661 jobs/registry.go:1538  [T1,n1] 1060  AUTO SPAN CONFIG RECONCILIATION job 855672065312325633: stepping through state succeeded
I230411 08:13:12.111084 9693 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=system-config-cache] 1057  MuxRangeFeed  terminating after 1.029757453s with err=recv msg error: recv: context canceled
I230411 08:13:12.111182 10126 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=tenant-settings-watcher] 1058  MuxRangeFeed  terminating after 4.176925ms with err=recv msg error: recv: context canceled
I230411 08:13:12.111324 9053 kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:80  [T1,n1,rangefeed=table-stats-cache] 1061  MuxRangeFeed  terminating after 1.085758969s with err=recv msg error: recv: context canceled
E230411 08:13:12.111422 9661 jobs/registry.go:943  [T1,n1] 1062  error getting live session: node unavailable; try another peer
W230411 08:13:12.116890 10691 jobs/adopt.go:518  [T1,n1] 1063  could not clear job claim: clear-job-claim: failed to send RPC: sending to all replicas failed; last error: ba: Scan [/Table/15/1/102,/Table/15/1/103), [txn: 784dff28], [can-forward-ts] RPC error: node unavailable; try another peer
W230411 08:13:12.121307 8581 server/server_sql.go:1696  [T1,n1] 1064  server shutdown without a prior graceful drain
--- FAIL: TestErrorDuringExtendedProtocolCommit (1.36s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-26868

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9634705?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9634705?buildTab=artifacts#/) on master @ [e7cac13f62d69373bc3a025d38de03ea2a02ee35](https://github.com/cockroachdb/cockroach/commits/e7cac13f62d69373bc3a025d38de03ea2a02ee35):


```
I230416 17:59:14.103307 10033 jobs/registry.go:1543  [T1,n1] 954  MIGRATION job 857202865141514241: stepping through state running
I230416 17:59:14.115166 10033 jobs/registry.go:613  [T1,n1,job=MIGRATION id=857202865141514241,upgrade=1000022.2-90] 955  active version is 1000023.1-2
I230416 17:59:14.125384 10518 jobs/adopt.go:261  [T1,n1,job=MIGRATION id=857202865141514241,upgrade=1000022.2-90] 956  job 102: resuming execution
I230416 17:59:14.126824 10033 jobs/registry.go:1543  [T1,n1] 957  MIGRATION job 857202865141514241: stepping through state succeeded
I230416 17:59:14.139701 10562 jobs/registry.go:1543  [T1,n1] 958  AUTO CONFIG RUNNER job 102: stepping through state running
I230416 17:59:14.147079 8486 jobs/wait.go:145  [T1,n1] 959  waited for 1 [857202865141514241] queued jobs to complete 58.125047ms
I230416 17:59:14.147278 8486 upgrade/upgrademanager/manager.go:283  [T1,n1] 960  running permanent upgrade for version 1000022.2-98
I230416 17:59:14.147326 10562 server/autoconfig/auto_config.go:97  [T1,n1,job=AUTO CONFIG RUNNER id=102] 961  using provider with type acprovider.NoTaskProvider
I230416 17:59:14.147371 10562 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 962  waiting for environment activation...
I230416 17:59:14.147404 10562 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 963  waiting for environment activation...
I230416 17:59:14.174263 8486 jobs/registry.go:613  [T1,n1] 964  active version is 1000023.1-2
I230416 17:59:14.182752 10644 2@rpc/context.go:2565  [T1,n1,rnode=1,raddr=127.0.0.1:35811,class=default,rpc] 965  connection is now ready
I230416 17:59:14.184838 8486 upgrade/upgrademanager/manager.go:742  [T1,n1] 966  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230416 17:59:14.184913 10479 jobs/adopt.go:261  [T1,n1] 967  job 857202865429970945: resuming execution
I230416 17:59:14.200186 10666 jobs/registry.go:1543  [T1,n1] 968  MIGRATION job 857202865429970945: stepping through state running
I230416 17:59:14.213693 10635 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857202865429970945,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 969 ={"Timestamp":1681667954209423265,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230416 17:59:14.218808 10711 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857202865429970945,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 970 ={"Timestamp":1681667954214093798,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230416 17:59:14.222525 10688 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857202865429970945,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 971 ={"Timestamp":1681667954219141411,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230416 17:59:14.228179 10743 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=857202865429970945,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 972 ={"Timestamp":1681667954223634392,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230416 17:59:14.230023 10666 jobs/registry.go:1543  [T1,n1] 973  MIGRATION job 857202865429970945: stepping through state succeeded
I230416 17:59:14.247903 8486 jobs/wait.go:145  [T1,n1] 974  waited for 1 [857202865429970945] queued jobs to complete 63.017005ms
I230416 17:59:14.247953 8486 server/server_sql.go:1618  [T1,n1] 975  done ensuring all necessary startup migrations have run
I230416 17:59:14.248286 10770 jobs/job_scheduler.go:407  [T1,n1] 976  waiting 4m0s before scheduled jobs daemon start
I230416 17:59:14.248618 10776 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 977  tenant-settings-watcher: established range feed cache
I230416 17:59:14.249203 10778 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 978  tenant-capability-watcher: established range feed cache
I230416 17:59:14.249615 10778 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 979  received results of a full table scan for tenant capabilities
I230416 17:59:14.252003 10774 server/auto_upgrade.go:78  [T1,n1] 980  no need to upgrade, cluster already at the newest version
I230416 17:59:14.255047 10772 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 981  warmed privileges for virtual tables in 6.711715ms
I230416 17:59:14.281609 8756 3@pebble/event.go:717  [n1,s1,pebble] 982  [JOB 7] WAL created 000006
I230416 17:59:14.285362 8486 1@server/server_sql.go:1736  [T1,n1] 983  serving sql connections
I230416 17:59:14.286624 8486 testutils/testcluster/testcluster.go:1375  [-] 984  WaitForFullReplication
I230416 17:59:14.286661 8486 testutils/testcluster/testcluster.go:1379  [-] 985  WaitForFullReplication took: 336ns
    autocommit_extended_protocol_test.go:197: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/tests/autocommit_extended_protocol_test.go:197
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
I230416 17:59:14.298003 8486 testutils/testcluster/testcluster.go:146  [-] 986  TestCluster quiescing nodes
I230416 17:59:14.298238 8513 sql/stats/automatic_stats.go:550  [T1,n1] 987  quiescing auto stats refresher
W230416 17:59:14.298272 9852 kv/kvserver/intentresolver/intent_resolver.go:826  [-] 988  failed to gc transaction record: could not GC completed transaction anchored at /Table/53/1/857202863216558081/"legacy_progress": node unavailable; try another peer
I230416 17:59:14.298452 8677 server/start_listen.go:103  [T1,n1] 989  server shutting down: instructing cmux to stop accepting
I230416 17:59:14.298868 10644 rpc/context.go:2302  [T1,n1,rnode=1,raddr=127.0.0.1:35811,class=default,rpc] 990  connection heartbeat loop ended with err: <nil>
W230416 17:59:14.298861 8896 sql/sqlliveness/slinstance/slinstance.go:334  [T1,n1] 991  exiting heartbeat loop
I230416 17:59:14.298936 9670 jobs/registry.go:1543  [T1,n1] 992  AUTO SPAN CONFIG RECONCILIATION job 857202862321434625: stepping through state succeeded
W230416 17:59:14.298965 8896 sql/sqlliveness/slinstance/slinstance.go:321  [T1,n1] 993  exiting heartbeat loop with error: node unavailable; try another peer
E230416 17:59:14.299010 8896 server/server_sql.go:508  [T1,n1] 994  failed to run update of instance with new session ID: node unavailable; try another peer
E230416 17:59:14.299044 10562 jobs/registry.go:948  [T1,n1] 995  error getting live session: node unavailable; try another peer
I230416 17:59:14.303699 8486 server/server_controller_orchestration.go:263  [T1,n1] 996  server controller shutting down ungracefully
I230416 17:59:14.303733 8486 server/server_controller_orchestration.go:274  [T1,n1] 997  waiting for tenant servers to report stopped
W230416 17:59:14.303838 8486 server/server_sql.go:1696  [T1,n1] 998  server shutdown without a prior graceful drain
--- FAIL: TestErrorDuringExtendedProtocolCommit (1.23s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101558 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-sessions branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestErrorDuringExtendedProtocolCommit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9687691?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9687691?buildTab=artifacts#/) on master @ [3df495f70198e0fdd2bfc96ae5483e78fcb1ed02](https://github.com/cockroachdb/cockroach/commits/3df495f70198e0fdd2bfc96ae5483e78fcb1ed02):


```
I230419 17:46:15.673347 10492 jobs/registry.go:613  [T1,n1,job=MIGRATION id=858049660892839937,upgrade=1000022.2-90] 827  active version is 1000023.1-2
I230419 17:46:15.681823 10529 jobs/adopt.go:261  [T1,n1,job=MIGRATION id=858049660892839937,upgrade=1000022.2-90] 828  job 102: resuming execution
I230419 17:46:15.682970 10492 jobs/registry.go:1543  [T1,n1] 829  MIGRATION job 858049660892839937: stepping through state succeeded
I230419 17:46:15.695061 10510 jobs/registry.go:1543  [T1,n1] 830  AUTO CONFIG RUNNER job 102: stepping through state running
I230419 17:46:15.702538 8426 jobs/wait.go:145  [T1,n1] 831  waited for 1 [858049660892839937] queued jobs to complete 57.47455ms
I230419 17:46:15.702707 8426 upgrade/upgrademanager/manager.go:283  [T1,n1] 832  running permanent upgrade for version 1000022.2-98
I230419 17:46:15.703528 10510 server/autoconfig/auto_config.go:97  [T1,n1,job=AUTO CONFIG RUNNER id=102] 833  using provider with type acprovider.NoTaskProvider
I230419 17:46:15.703576 10510 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 834  waiting for environment activation...
I230419 17:46:15.703601 10510 server/autoconfig/auto_config.go:74  [T1,n1,job=AUTO CONFIG RUNNER id=102] 835  waiting for environment activation...
I230419 17:46:15.723309 8426 jobs/registry.go:613  [T1,n1] 836  active version is 1000023.1-2
I230419 17:46:15.730785 8426 upgrade/upgrademanager/manager.go:742  [T1,n1] 837  running Upgrade to 1000022.2-98: "change TTL for SQL Stats system tables"
I230419 17:46:15.730858 10201 jobs/adopt.go:261  [T1,n1] 838  job 858049661182738433: resuming execution
I230419 17:46:15.741637 10538 2@rpc/context.go:2565  [T1,n1,rnode=1,raddr=127.0.0.1:35985,class=default,rpc] 839  connection is now ready
I230419 17:46:15.745450 10542 jobs/registry.go:1543  [T1,n1] 840  MIGRATION job 858049661182738433: stepping through state running
I230419 17:46:15.757667 10649 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=858049661182738433,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 841 ={"Timestamp":1681926375753960784,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":42,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230419 17:46:15.761836 10606 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=858049661182738433,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 842 ={"Timestamp":1681926375758009307,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_statistics CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":43,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_statistics","Options":["\"gc.ttlseconds\" = 3600"]}
I230419 17:46:15.764983 10690 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=858049661182738433,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 843 ={"Timestamp":1681926375762147554,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.statement_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":60,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.statement_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230419 17:46:15.767691 10709 1@util/log/event_log.go:32  [T1,n1,job=MIGRATION id=858049661182738433,upgrade=1000022.2-98,intExec=set-SQLStatsTables-TTL] 844 ={"Timestamp":1681926375765296416,"EventType":"set_zone_config","Statement":"ALTER TABLE \"\".system.transaction_activity CONFIGURE ZONE USING \"gc.ttlseconds\" = $1","Tag":"CONFIGURE ZONE","User":"root","DescriptorID":61,"ApplicationName":"$ internal-set-SQLStatsTables-TTL","PlaceholderValues":["3600"],"Target":"TABLE system.public.transaction_activity","Options":["\"gc.ttlseconds\" = 3600"]}
I230419 17:46:15.769284 10542 jobs/registry.go:1543  [T1,n1] 845  MIGRATION job 858049661182738433: stepping through state succeeded
I230419 17:46:15.789528 8426 jobs/wait.go:145  [T1,n1] 846  waited for 1 [858049661182738433] queued jobs to complete 58.694839ms
I230419 17:46:15.789579 8426 server/server_sql.go:1618  [T1,n1] 847  done ensuring all necessary startup migrations have run
I230419 17:46:15.789939 10591 jobs/job_scheduler.go:407  [T1,n1] 848  waiting 2m0s before scheduled jobs daemon start
I230419 17:46:15.790229 10757 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 849  tenant-settings-watcher: established range feed cache
I230419 17:46:15.790879 10759 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 850  tenant-capability-watcher: established range feed cache
I230419 17:46:15.791264 10759 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:149  [T1,n1] 851  received results of a full table scan for tenant capabilities
I230419 17:46:15.795063 10755 server/auto_upgrade.go:78  [T1,n1] 852  no need to upgrade, cluster already at the newest version
I230419 17:46:15.799685 10593 sql/syntheticprivilegecache/cache.go:199  [T1,n1] 853  warmed privileges for virtual tables in 9.729631ms
I230419 17:46:15.829510 8758 3@pebble/event.go:717  [n1,s1,pebble] 854  [JOB 7] WAL created 000006
I230419 17:46:15.832252 8426 1@server/server_sql.go:1736  [T1,n1] 855  serving sql connections
I230419 17:46:15.833819 8426 testutils/testcluster/testcluster.go:1375  [-] 856  WaitForFullReplication
I230419 17:46:15.833876 8426 testutils/testcluster/testcluster.go:1379  [-] 857  WaitForFullReplication took: 390ns
    autocommit_extended_protocol_test.go:197: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/tests/autocommit_extended_protocol_test.go:197
        	Error:      	An error is expected but got nil.
        	Test:       	TestErrorDuringExtendedProtocolCommit
I230419 17:46:15.848568 8426 testutils/testcluster/testcluster.go:146  [-] 858  TestCluster quiescing nodes
I230419 17:46:15.848685 9136 sql/stats/automatic_stats.go:550  [T1,n1] 859  quiescing auto stats refresher
W230419 17:46:15.848906 10183 kv/kvserver/intentresolver/intent_resolver.go:826  [-] 860  failed to gc transaction record: could not GC completed transaction anchored at /Table/3/1/102/2/1: node unavailable; try another peer
I230419 17:46:15.848932 8681 server/start_listen.go:103  [T1,n1] 861  server shutting down: instructing cmux to stop accepting
I230419 17:46:15.849262 10538 rpc/context.go:2302  [T1,n1,rnode=1,raddr=127.0.0.1:35985,class=default,rpc] 862  connection heartbeat loop ended with err: <nil>
I230419 17:46:15.849318 9650 jobs/registry.go:1543  [T1,n1] 863  AUTO SPAN CONFIG RECONCILIATION job 858049658080657409: stepping through state succeeded
W230419 17:46:15.849317 9090 sql/sqlliveness/slinstance/slinstance.go:334  [T1,n1] 864  exiting heartbeat loop
W230419 17:46:15.849579 9090 sql/sqlliveness/slinstance/slinstance.go:321  [T1,n1] 865  exiting heartbeat loop with error: node unavailable; try another peer
E230419 17:46:15.849620 9090 server/server_sql.go:508  [T1,n1] 866  failed to run update of instance with new session ID: node unavailable; try another peer
W230419 17:46:15.855770 9650 jobs/adopt.go:518  [T1,n1] 867  could not clear job claim: clear-job-claim: node unavailable; try another peer
W230419 17:46:15.855775 10510 jobs/adopt.go:518  [T1,n1] 868  could not clear job claim: clear-job-claim: failed to send RPC: sending to all replicas failed; last error: ba: Scan [/Table/15/1/102,/Table/15/1/103), [txn: 77d182df], [can-forward-ts] RPC error: node unavailable; try another peer
I230419 17:46:15.860443 8426 server/server_controller_orchestration.go:263  [T1,n1] 869  server controller shutting down ungracefully
I230419 17:46:15.860477 8426 server/server_controller_orchestration.go:274  [T1,n1] 870  waiting for tenant servers to report stopped
W230419 17:46:15.860550 8426 server/server_sql.go:1696  [T1,n1] 871  server shutdown without a prior graceful drain
--- FAIL: TestErrorDuringExtendedProtocolCommit (1.22s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101673 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-queries T-sql-sessions branch-release-23.1.0]
- #101558 sql/tests: TestErrorDuringExtendedProtocolCommit failed [C-test-failure O-robot T-sql-sessions branch-release-23.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestErrorDuringExtendedProtocolCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/tests/autocommit_extended_protocol_test.go
+++ b/pkg/sql/tests/autocommit_extended_protocol_test.go
@@ -170,12 +170,12 @@ func TestErrorDuringExtendedProtocolCommit(t *testing.T) {
 		},
 		BeforeAutoCommit: func(ctx context.Context, stmt string) error {
 			if shouldErrorOnAutoCommit.Get() {
-				shouldErrorOnAutoCommit.Set(false)
 				// Only inject the error if we're in the same trace as the one we
 				// saw when executing our test query. This is so we know that this
 				// autocommit corresponds to our test qyery rather than an internal
 				// query.
 				if traceID == tracing.SpanFromContext(ctx).TraceID() {
+					shouldErrorOnAutoCommit.Set(false)
 					return errors.New("injected error")
 				}
 			}

